Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM node:8.16-alpine as builder
+
+# Install SSL ca certificates
+#RUN apk update && apk add git && apk add ca-certificates
+
+# Create appuser
+#RUN adduser -D -g '' appuser && mkdir /go-analyzer
+RUN mkdir /javascript-analyzer
+
+# source code
+WORKDIR /javascript-analyzer
+
+# get the rest of the source code
+COPY . /javascript-analyzer
+
+# build
+RUN yarn install && yarn build
+
+# Build a minimal and secured container
```

Comment:
Little skeptical about this multiple builds. How minimal and secure is it compared to simple single build?

Reply:
It makes more sense in other languages where you have a build sdk and a runtime environment: in the minimal container you distribute only the runtime environment and the compiled version.

For dotnetcore images the size difference  is one order of magnitude: ~1Gb versus 100 Mb

Another advantage is that you do not distribute the source code but only the compiled version.

Since this is javascript I don't know if it matters. I'm open to suggestions

Reply:
> It makes more sense in other languages where you have a build sdk and a runtime environment: in the minimal container you distribute only the runtime environment and the compiled version.

This makes total sense. 👍 

> Since this is javascript I don't know if it matters. I'm open to suggestions

Same. My comment was based on the script copying most files (including node_modules) into the next container. It does not make much difference for uncompiled languages like JS/Ruby IMO.

Reply:
> My comment was based on the script copying most files (including node_modules)

Perhaps we only copy `dist` ?